### PR TITLE
Create SplunkUtills

### PR DESCRIPTION
**Pull request by EyalBerger**
I uploaded a first version of SplunkUtills. **It's ready to merge now**.

**How to test** - download SplunkUtills.py, credentials.json template and Getting started with SplunkUtills.ipynb files to the same folder. Then set your credentials and run the notebook.

Note that I compared my API results (when using a loop) to the Splunk app results, and founded a gap regarding the last day of the search (I send both queries in 11.5.2019, aiming to get last 30 days data - only the app returned full data from 11.4.2019).

### CODE REVIEW

File: splunk/SplunkUtills/SplunkUtills.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    Returns:
+        DataFrame: query results.
+    '''
+
+    print('start..\n')
+
+    print('your query:\n {}\n'.format("|\n".join(query.split("|"))))
+
+    # imports
```
Comment (cjer) [inline splunk/SplunkUtills/SplunkUtills.py]: I would move imports outside (beginning of file)

File: splunk/SplunkUtills/SplunkUtills.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    print('imports completed\n')
+
+    # connect to splunk
+    service = client.connect(host=host, port=port, username=username, password=password)
+    print('connection succeed\n')
+
+    # query splunk
+    kwargs_normalsearch = {"exec_mode": "normal", "count": 0}
+    job = service.jobs.create(query, **kwargs_normalsearch)
+
```
Comment (cjer) [inline splunk/SplunkUtills/SplunkUtills.py]: I think we should add a time limit (with some reasonable default, a few minutes), and cancel the job when it is reached.

File: splunk/SplunkUtills/credentials.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"host": "HOST", "port": PORT, "username": "USERNAME", "password": "PASSWORD"}
```
Comment (cjer) [inline splunk/SplunkUtills/credentials.json]: You should state that the PORT that is needed here is the splunk management port and not web port.
This should be added in the notebook and module README.

**Comment by EyalBerger** [comment]
Thanks for the comments. I will update my code and upload a new version, taking in a account your comments.
As for your questions - 

1. I probably misunderstood this issue when we talked about Splunk syntax. I thought that indexes order is matter when filtering, e.g:
`search index=siri earliest=-7d latest=now route_short_name=501 planned_start_time=20:00:00 |
 fields timestamp, agency_id, route_id, route_short_name, service_id, planned_start_time, bus_id, predicted_end_time, time_recorded, lat, lon |`
vs - 
`search planned_start_time=20:00:00 route_short_name=501 index=siri earliest=-7d latest=now  |
 fields timestamp, agency_id, route_id, route_short_name, service_id, planned_start_time,  bus_id, predicted_end_time, time_recorded, lat, lon |`  
But actually, running both queries takes the same time. I will remove this comment.
2. I have checked it  now. Most of the time goes for running the job and for reading the job results.

File: splunk/SplunkUtills/SplunkUtills.py (hunk 1 above)
Comment (EyalBerger) [reply splunk/SplunkUtills/SplunkUtills.py]: Done

File: splunk/SplunkUtills/SplunkUtills.py (hunk 2 above)
Comment (EyalBerger) [reply splunk/SplunkUtills/SplunkUtills.py]: Done

File: splunk/SplunkUtills/credentials.json (hunk 3 above)
Comment (EyalBerger) [reply splunk/SplunkUtills/credentials.json]: Done

**Comment by EyalBerger** [comment]
I updated SplunkUtills.py and Getting started with SplunkUtills.ipynb according to your helpful comments. Next Monday I will do last tests for the new version and add a README file.

**Comment by EyalBerger** [comment]
I added a README file and did some tests. It's ready to merge.

**Pull request merged**